### PR TITLE
feat: add example workflows

### LINKED ISSUE #17: Full example of a workflow?
### Question

https://github.com/google-github-actions/deploy-cloudrun#usage appears to be a snippet. I am trying to convert over https://github.com/kaihendry/count/blob/gcp-cloudrun/.github/workflows/cloud-run.yml so I am looking for a full example.

Also steps to get the secret GCP key into Github would be nice too please.

### PR DESCRIPTION
**Pull request by averikitsch**
Fixes #17

### CODE REVIEW

File: .github/workflows/example-workflow-quickstart.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    - name: Deploy to Cloud Run
+      id: deploy
+      uses: google-github-actions/deploy-cloudrun@v0.2.0
+      with:
+        image: gcr.io/cloudrun/hello
+        service: hello-cloud-run
+        credentials: ${{ secrets.GCP_SA_KEY }}
+
+    - name: Show Output
+      run: echo ${{ steps.deploy.outputs.url }}
```
Comment (bharathkkb) [inline .github/workflows/example-workflow-quickstart.yaml:36]: nit: newline

File: README.md [hunk 2]
```diff
@@ -174,9 +174,93 @@ only works using a custom runner hosted on GCP.**
...
+    service_account_key: ${{ secrets.GCP_SA_KEY }}
+
+- name: Deploy to Cloud Run
+  run: |-
+    gcloud run deploy $SERVICE \
+      --region $REGION \
+      --image gcr.io/$PROJECT_ID/$SERVICE \
+      --platform managed \
+      --set-env-vars NAME="Hello World" \
+      --allow-unauthenticated
```
Comment (bharathkkb) [inline README.md:243]: Should we include the  `--allow-unauthenticated` flag if native action does not set it?

File: .github/workflows/example-workflow.yaml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    - name: Deploy to Cloud Run
+      id: deploy
+      uses: google-github-actions/deploy-cloudrun@v0.2.0
+      with:
+        service: ${{ env.SERVICE }}
+        image: gcr.io/${{ env.PROJECT_ID }}/${{ env.SERVICE }}:${{  github.sha }}
+        region: ${{ env.REGION }}
+
+    - name: Show Output
+      run: echo ${{ steps.deploy.outputs.url }}
```
Comment (bharathkkb) [inline .github/workflows/example-workflow.yaml:57]: nit: newline

File: .github/workflows/example-workflow.yaml [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+on:
+  push:
+    branches:
+      - example-build
```
Comment (bharathkkb) [inline .github/workflows/example-workflow.yaml:18]:
```suggestion
      - example-build-deploy
```

File: README.md [hunk 5]
```diff
@@ -174,9 +174,93 @@ only works using a custom runner hosted on GCP.**
...
+
+1.  Add the following [secrets to your repository's secrets][gh-secret]:
+
+    - `GCP_PROJECT`: Google Cloud project ID
+
+    - `GCP_SA_KEY`: the downloaded service account key
+
+### Deploy a prebuilt container
+
+To run this workflow, push to the branch named `example-deploy`:
```
Comment (bharathkkb) [inline README.md:209]: Should we link the corresponding workflow file here so users can review?

File: README.md [hunk 6]
```diff
@@ -174,9 +174,93 @@ only works using a custom runner hosted on GCP.**
...
+
+1.  [Create a Google Cloud service account][sa] or select an existing one.
+
+1.  Add the the following [Cloud IAM roles][roles] to your service account:
+
+    - `Cloud Run Admin` - allows for the creation of new Cloud Run services
+
+    - `Service Account User` -  required to deploy to Cloud Run as service account
+
+    - `Storage Admin` - allow push to Google Container Registry
```
Comment (bharathkkb) [inline README.md:197]: This is broad at the project level, perhaps we can link to further reading [here](https://cloud.google.com/container-registry/docs/access-control#grant) for best practices of granting at the bucket level.

File: .github/workflows/deploy-cloudrun-credentials-it.yml [hunk 7]
```diff
@@ -1,6 +1,9 @@
 name: deploy-cloudrun credentials Integration
 
-on: [push]
+on:
+  push:
+    branches-ignore: 
+    - 'example-*'
```
Comment (bharathkkb) [inline .github/workflows/deploy-cloudrun-credentials-it.yml:6]: Do we need to ignore these branches?

**Review by bharathkkb — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (averikitsch) [reply README.md:243]: sgtm

File: .github/workflows/deploy-cloudrun-credentials-it.yml (hunk 7 above)
Comment (averikitsch) [reply .github/workflows/deploy-cloudrun-credentials-it.yml:6]: I think it's nice that if someone is trying to run the example workflows it doesn't kick off this test workflow.

File: README.md (hunk 5 above)
Comment (averikitsch) [reply README.md:209]: sgtm

File: README.md (hunk 6 above)
Comment (averikitsch) [reply README.md:197]: sgtm

**Pull request merged**